src/Build/BackEnd/Components/SdkResolution/CachingSdkResolverService.cs

```diff
@@ -53,7 +55,7 @@ public override SdkResult ResolveSdk(int submissionId, SdkReference sdk, Logging
                  */
                 result = cached.GetOrAdd(
                     sdk.Name,
-                    key => base.ResolveSdk(submissionId, sdk, loggingContext, sdkReferenceLocation, solutionPath, projectPath, interactive));
+                    key => base.ResolveSdk(submissionId, sdk, loggingContext, sdkReferenceLocation, solutionPath, projectPath, interactive, globalProperties));
```

Comment:
This is probably a correctness argument for not passing global properties. Sdks are currently cached per build submission and not per build request (a build submission has multiple requests). We'd have to redo their caching to change the key to be a Microsoft.Build.BackEnd.ConfigurationMetadata. But by doing this we'd loose perf, and I don't see what we gain for it.